src/partition/ReceivedPartition.cpp

```diff
@@ -207,17 +211,21 @@ void ReceivedPartition::compute()
       PRECICE_DEBUG("Send partition feedback to primary rank");
       utils::IntraComm::getCommunication()->sendRange(vertexIDs, 0);
     } else { // Primary
-      int              numberOfVertices = _mesh->vertices().size();
-      std::vector<int> vertexIDs(numberOfVertices, -1);
+
+      mesh::Mesh::VertexDistribution vertexDistribution;
+      int                            numberOfVertices = _mesh->vertices().size();
+      std::vector<int>               vertexIDs(numberOfVertices, -1);
       for (int i = 0; i < numberOfVertices; i++) {
         vertexIDs[i] = _mesh->vertices()[i].getGlobalIndex();
       }
-      _mesh->getVertexDistribution()[0] = std::move(vertexIDs);
+      vertexDistribution[0] = std::move(vertexIDs);
```

Comment:
Do we need an assert for `vertexDistribution.empty()`?

Reply:
Same as above.